### ISSUE TITLE
A new fork req. supporting A13

### ISSUE THREAD

**Issue by alexanderrogling** [sn-00-x/aa4mg#3]
Hello sn-00-X I love that you have made a systemless AA module and used it on LinageOS 19.

But the thing is that I have upgrade my Moto G7 Plus to LinageOS 20 and did not think about this module (but did think about all other modules I use funny enough)

So now when I try connect any car I have access to with AA, it just straight crashes.

Would love if you could make this module for A13 and I'm willing to test it (if you are sure enough it will not cause bootloo, if that is a big risk I need to know so I could use a phone that is only collecting dust)

I should be able to try and make a firm my self but I have not access to enough powerful PC to be sustanible to re-do the work time and time again, that is why I'm asking if you are willing to make a fork, even if you don't have access to any physical device running LinageOS 20.

Thanks again for this great module it worked like how anyone could want it to work.

**Comment by byReqz** [comment]
I have updated the module for Android 13 in #4 . Please check if it works for you.

**Comment by alexanderrogling** [comment]
Thank you, was seeing the e-mail today where GitGub reported that you have answer, I will try tomorrow about (5:50 my time living in Sweden) and report back to you as soon as I have opportunity or after the work tomorrow (Saturday).

**Comment by alexanderrogling** [comment]
I can report that you have made at least one error you have created a directory called aa4mg-0.2.a12-pre inside if the zip file which makes Magisk return error "Unzip error" when extracted and zip without the directory it at least installs it.

I will as said previous try it tomorrow...

Will hope fully even return a yay or no about latest 6:30 my time

**Comment by byReqz** [comment]
The thing youre describing seems like the "download as zip" feature of Github on the wrong branch. That'll create a folder with the branch name inside of the zip.

You can build the module as per the instructions in the readme:
```
git clone https://github.com/byReqz/aa4mg
cd aa4mg
zip aa4mg.zip -9r *
```

once this is merged im sure sn-00-x will create a proper release aswell

**Comment by alexanderrogling** [comment]
Using it as it's when extracting the zip file and re-zipping it with out the map as I said before, will at least give me Communication error 8 even as I have give AA all the privileges.

Can't try as you have give me tips to do before this afternoon sometime, updating AA to latest version will instead cause it to return the error related to be not be installed from factory I will try some time later today before trying your idea if fixing the unzip error to update AA to  8.2.623924 to see if that changes anything.

By the way when I have try updating it I will only edit this post to not make this thread a metric mile long :)

**Comment by byReqz** [comment]
The straight up crashing is a symptom of the A12 version of the module running on A13. Please verify that all the new files are actually in `/system/product`. Would be interesting to see if building the module as intended is different to what you did as im not sure how permissions are preserved that way.

**Comment by alexanderrogling** [comment]
When I did clone the repo directly on the phone and zip the files it "kind of" worked directly I was req. to replace the included with 8.2.623924-releas (I assume it will work with newer also) to get passed the "setup screen" that is shown on first start.

**Comment by sn-00-x** [comment]
Thanks again for your work @byReqz. I've merged your pull request!
